simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,787 @@
+                          str(multiplier*particle.coordinates[1])+' '+str(multiplier*particle.coordinates[2]) + '\n'
+                    f.write(line)
+
+            except:
+                    ('error in write block of WriteEspressoInputFile')
+                    raise
+        print('finished writing file')
+        f.closed
+
+    def kluge(self,f):
+        line =' C 1.0 2.0 3.0 \n'
+        f.write(line)
+        line =' C 3.0 3.0 4.0 \n'
+        f.write(line)
+        line =' C 3.0 4.0 5.0 \n'
+        f.write(line)
+        line =' C 4.0 5.0 6.0 \n'
+        f.write(line)
+        return
+
+    def WriteEspressoPPFile(self,ppfilename="testpp.in"):
+        '''
+        this writes an auxiliary required file determined the plot parameters
+        :return:
+        '''
+        outdir = './'
+        plotfile = 'output.charge'
+        outfile = 'density.dat'
+        lines=['&inputpp',
+               'prefix =\'qe_output\'',
+               'filplot=\''+plotfile+'\'',
+               'plot_num=0',
+               'outdir = '+outdir,
+               '/',
+                '&plot',
+               'nfile = 1',
+                'filepp(1) = \''+plotfile+'\'',
+               'weight(1) = 1.0',
+                'iflag = 3',
+                'output_format = 6',
+                'fileout = \''+outfile+'\''  ]
+        with open(ppfilename,'w') as pp:
+            for line in lines:
+                print'line:'+str(line)
+                pp.write(str(line)+'\n')
+        print('finished writing file')
+
+    def ReadEspressoInputFile(self,file_name):
+        """  This class parses  Espresso data files, either input or output
+        (produced by the espresso command  write_data) and calls a handler
+        which processes the parsed information.
+        A handler class is given the parsed information. This handler class can
+        then determine what to  do with it.  For, example it could just store
+        the data in memory (see LammpsSimpleDataHandler) or write it some other
+        data file (e.g. a CUDS-file).
+        cheatsheet for quantum espresso
+
+        0. create/obtain input file such as pp.in from cuds data
+        1. run  (using mpi for instance )
+            for file describing simulation
+                mpirun -np 48 /usr/local/espresso/bin/pw.x < input_pw.in > pw.out &
+            once simulation is done, run on file describing desired output
+                mpirun -np 48 /usr/local/espresso/bin/pp.x < input_pp.in > pp.out &
+        2. convert output (which is a charge density file) into simphony format (see charge_density_xyz.cpp)
+
+        Parameters
+        ----------
+        handler :
+           handler will handle the parsed information provided by this class
+            SP[CUBA.TORQUE] = calculation_type
+            SP[CUBA.ZETA_POTENTIAL] = restart_mode
+            SP[CUBA.YOUNG_MODULUS] = pseudo_dir
+            SP[CUBA.VOLUME_FRACTION] = prefix
```

Comment:
please take into account that a number of separate QE wrappers may be "active" in SimPhoNy, so each wrapper instance should have a different working directory. 
The wrapper configures and start QE to perform a run (e.g., self consistent calculation), then waits for further instructions from the user/script. Examples: get/set attributes, move atoms, etc, or start additional calculations of forces, charge distributions, etc. The wrapper keeps all this time (until the function is out of scope) the working directory alive. The SSB discusses currently whether a terminate method for the wrappers is needed.